crates/proto/src/tcp/tcp_stream.rs

```diff
@@ -35,7 +35,7 @@ pub trait DnsTcpStream: AsyncRead + AsyncWrite + Unpin + Send + Sync + Sized + '
 #[async_trait]
 pub trait Connect: DnsTcpStream {
     /// connect to tcp
-    async fn connect(addr: SocketAddr) -> io::Result<Self>;
+    async fn connect(addr: SocketAddr, bind_addr: Option<SocketAddr>) -> io::Result<Self>;
```

Comment:
For this one, I think we should keep the original `connect` and have a default implementation that passes in a None to the `bind_addr`. Then rename this method to `connect_with_bind`.

Reply:
Done